test/unit/adapters/uniswap-v3-adapter.spec.ts

```diff
@@ -256,43 +268,125 @@ describe('UniswapV3Adapter', () => {
+        given(async () => {
+          setCurrentCardinality(pool1, TARGET_CARDINALITY);
+          setCurrentCardinality(pool2, TARGET_CARDINALITY);
+          tx = await adapter.internalAddOrModifySupportForPair(TOKEN_A, TOKEN_B, []);
+        });
+        thenPoolIsNotIncreased(() => pool1);
+        thenPoolIsNotIncreased(() => pool2);
+        thenPoolsAreStoredAndEventIsEmitted(() => ({ tx, pools: [pool1, pool2] }));
       });
-      then('event is emitted', async () => {
-        await expect(tx).to.emit(adapter, 'UpdatedSupport').withArgs(TOKEN_A, TOKEN_B, [pool1.address, pool2.address]);
+    });
+
+    when('there is enough gas to initialize one pool', () => {
+      describe('and the first one needs initializing', () => {
+        let tx: TransactionResponse;
+        given(async () => {
+          setCurrentCardinality(pool1, 0);
+          setCurrentCardinality(pool2, 0);
+          tx = await adapter.internalAddOrModifySupportForPair(TOKEN_A, TOKEN_B, [], { gasLimit: gasForOnePool });
+        });
+        thenPoolIsIncreased(() => pool1);
+        thenPoolIsNotIncreased(() => pool2);
+        thenPoolsAreStoredAndEventIsEmitted(() => ({ tx, pools: [pool1] }));
+      });
+      describe('and the first is already initialized', () => {
+        let tx: TransactionResponse;
+        given(async () => {
+          setCurrentCardinality(pool1, TARGET_CARDINALITY);
+          setCurrentCardinality(pool2, 0);
+          tx = await adapter.internalAddOrModifySupportForPair(TOKEN_A, TOKEN_B, [], { gasLimit: gasForOnePool });
+        });
+        thenPoolIsNotIncreased(() => pool1);
+        thenPoolIsIncreased(() => pool2);
+        thenPoolsAreStoredAndEventIsEmitted(() => ({ tx, pools: [pool1, pool2] }));
+      });
+      describe('and they are all initialized', () => {
+        let tx: TransactionResponse;
+        given(async () => {
+          setCurrentCardinality(pool1, TARGET_CARDINALITY);
+          setCurrentCardinality(pool2, TARGET_CARDINALITY);
+          tx = await adapter.internalAddOrModifySupportForPair(TOKEN_A, TOKEN_B, [], { gasLimit: gasForOnePool });
+        });
+        thenPoolIsNotIncreased(() => pool1);
+        thenPoolIsNotIncreased(() => pool2);
+        thenPoolsAreStoredAndEventIsEmitted(() => ({ tx, pools: [pool1, pool2] }));
+      });
+    });
+
+    when('there is not enough gas for pool initialization', () => {
+      describe('and they are all initialized', () => {
+        let tx: TransactionResponse;
+        given(async () => {
+          setCurrentCardinality(pool1, TARGET_CARDINALITY);
+          setCurrentCardinality(pool2, TARGET_CARDINALITY);
+          tx = await adapter.internalAddOrModifySupportForPair(TOKEN_A, TOKEN_B, [], { gasLimit: 1_000_000 });
+        });
+        thenPoolIsNotIncreased(() => pool1);
+        thenPoolIsNotIncreased(() => pool2);
+        thenPoolsAreStoredAndEventIsEmitted(() => ({ tx, pools: [pool1, pool2] }));
       });
     });
```

Comment:
Seems that we are missing the case in which they are not all initialized?

Reply:
That's on purpose, I'll add that test when we handle this situation on the adapter. For now, we only have a TODO